src/state/zustand/swapperStore/useSwapperStore.ts

```diff
@@ -0,0 +1,119 @@
+import type { KnownChainIds } from '@shapeshiftoss/types'
+import { DEFAULT_SLIPPAGE } from 'constants/constants'
+import { create } from 'zustand'
+import { devtools } from 'zustand/middleware'
+import { immer } from 'zustand/middleware/immer'
+import { TradeAmountInputField } from 'components/Trade/types'
+import type {
+  SetSwapperStoreAction,
+  SwapperAction,
+  SwapperStore,
+} from 'state/zustand/swapperStore/types'
+
+const createUpdateAction =
+  <T extends keyof SwapperStore>(set: SetSwapperStoreAction<SwapperStore>, key: T) =>
+  (value: SwapperStore[T]): void =>
+    set(() => ({ [key]: value }), false, {
+      type: `swapper/update${key.charAt(0).toUpperCase() + key.slice(1)}`,
+      value,
+    })
+
+export type SwapperState<T extends KnownChainIds = KnownChainIds> = SwapperStore<T> & SwapperAction
+export const useSwapperStore = (() => {
```

Comment:
Can we go without the `use` terminology to make it clear this isn't a hook? 🙏🏽

Reply:
Disregard me, `create` is react-specific and not the vanilla store
https://github.com/pmndrs/zustand/blob/962f83e01599257aa6eceaaa993965a0e296d359/src/react.ts#L92